### DISCUSSION TITLE
Why is the default storage orientation first_order for tensors?

### DISCUSSION THREAD

**Discussion by TimLauwers** [boostorg/ublas#174] [Q&A]
For matrices, the default is row-major order as I would expect with C(++) programming. However, in the documentation of tensors it is said that first_order corresponds to column-major order for matrices. Are there any performance improvements that first_order brings that I'm not aware of?

**Comment by bassoy** [comment]
> For matrices, the default is row-major order as I would expect with C(++) programming. However, in the documentation of tensors it is said that first_order corresponds to column-major order for matrices. Are there any performance improvements that first_order brings that I'm not aware of?

No. Not really. You can design fast and efficient matrix or tensor algorithms for any kind of linear storage format. Currently all tensor algorithms are sequential and not optimized.
